### PR TITLE
Add docker files

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- Add a Dockerfile and docker-compose config.
- At this time commissioning via Bluetooth doesn't work.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    image: matter-server:latest
+    container_name: matter-server
+    restart: always
+    security_opt:
+      # needed for bluetooth via dbus
+      - apparmor:unconfined
+    ports:
+      - "5580:5580"
+    volumes:
+      - ${USERDIR}/docker/matter-server/data:/data/
```
Comment (MartinHjelmare) [inline docker-compose.yml:17]:
```suggestion
      # Create an .env file that sets the USERDIR environment variable.
      - ${USERDIR}/docker/matter-server/data:/data/
```

**Comment by ZephireNZ** [comment]
Just used these for mine, worked a charm 👍

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    python3-pip \
+    python3-gi \
+    python3-gi-cairo \
+    python3-dbus \
+    python3-psutil \
+    unzip \
+    libcairo2 \
+    gdb \
+    git \
+    && git clone --depth 1 -b v1.0-branch \
```
Comment (marcelveldt) [inline Dockerfile:29]: we grab the certificates from master branch

Comment (MartinHjelmare) [reply Dockerfile:29]:
```suggestion
    && git clone --depth 1 -b master \
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
+FROM debian:bullseye as base
```
Comment (ludeeus) [inline Dockerfile:1]:
```suggestion
FROM debian:bullseye
```
Not used

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+# Set shell
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+ARG MATTER_SERVER_VERSION=1.0.7
+ARG HOME_ASSISTANT_CHIP_VERSION=2022.11.1
```
Comment (ludeeus) [inline Dockerfile:7]:
```suggestion
ARG \
    MATTER_SERVER_VERSION=1.0.7 \
    HOME_ASSISTANT_CHIP_VERSION=2022.11.1
```

File: Dockerfile (hunk 2 above)
Comment (ludeeus) [inline Dockerfile:29]:
```suggestion
    && git clone --depth 1 -b master \
```

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    && apt-get purge -y --auto-remove \
+    git \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /usr/src/*
+
+# hadolint ignore=DL3013
+RUN \
+    pip3 install \
+    home-assistant-chip-clusters==${HOME_ASSISTANT_CHIP_VERSION} \
+    home-assistant-chip-core==${HOME_ASSISTANT_CHIP_VERSION} \
```
Comment (ludeeus) [inline Dockerfile:42]:
```suggestion
    home-assistant-chip-clusters=="${HOME_ASSISTANT_CHIP_VERSION}" \
    home-assistant-chip-core=="${HOME_ASSISTANT_CHIP_VERSION}" \
```

**Review by marcelveldt — APPROVED** [review]
I'd say we go ahead and merge so we at least have a start.

**Pull request merged**